src/emmetHelper.ts

```diff
@@ -355,14 +355,22 @@ export function extractAbbreviation(document: TextDocument, position: Position,
 	let currentLine = getCurrentLine(document, position);
 	let currentLineTillPosition = currentLine.substr(0, position.character);
 	let lengthOccupiedByFilter = 0;
-	if (currentLineTillPosition.endsWith(`${filterDelimitor}${bemFilterSuffix}`)) {
-		lengthOccupiedByFilter = 4;
-		pos -= lengthOccupiedByFilter;
-		filter = bemFilterSuffix;
-	} else if (currentLineTillPosition.endsWith(`${filterDelimitor}${commentFilterSuffix}`)) {
-		lengthOccupiedByFilter = 2;
-		pos -= lengthOccupiedByFilter;
-		filter = commentFilterSuffix;
+	for (let i = 0; i < differentFilters; i++) {
```

Comment:
This whole logic is being repeated twice. It would be good to have this in a separate function which returns an object with `pos` and `filter`. `lengthOccupiedByFilter ` can then be calculated as `filter.length` + 1